File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/ev3lejosv1/strings.java
package generated.main;

import de.fhg.iais.roberta.runtime.*;
import de.fhg.iais.roberta.runtime.ev3.*;

import de.fhg.iais.roberta.mode.general.*;
import de.fhg.iais.roberta.mode.action.*;
import de.fhg.iais.roberta.mode.sensor.*;
import de.fhg.iais.roberta.mode.action.ev3.*;
import de.fhg.iais.roberta.mode.sensor.ev3.*;

import de.fhg.iais.roberta.components.*;

import java.util.LinkedHashSet;
import java.util.HashMap;
import java.util.Set;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

import lejos.remote.nxt.NXTConnection;

public class NEPOprog {
    private static Configuration brickConfiguration;

    private Set<UsedSensor> usedSensors = new LinkedHashSet<UsedSensor>();
    private Hal hal = new Hal(brickConfiguration, usedSensors);

    public static void main(String[] args) {
        try {
             brickConfiguration = new EV3Configuration.Builder()
                .setWheelDiameter(5.6)
                .setTrackWidth(18.0)
                .build();

            new NEPOprog().run();
        } catch ( Exception e ) {
            Hal.displayExceptionWaitForKeyPress(e);
        }
    }

    String ___text = "start:";
    String ___eight = " eight";
    float ___number = 0;

    public void run() throws Exception {
        // String Concat -- Start --
        ___text += String.valueOf(" one");
        if (!("start: one".equals(___text))) {
            System.out.println("Assertion failed: " + "POS-1" + "start: one" + "EQ" + ___text);
        }
        ___text += String.valueOf(String.valueOf(" two") + String.valueOf(" three"));
        if (!("start: one two three".equals(___text))) {
            System.out.println("Assertion failed: " + "POS-2" + "start: one two three" + "EQ" + ___text);
        }
        ___text += String.valueOf(String.valueOf(4) + String.valueOf(5));
        if (!("start: one two three45".equals(___text))) {
            System.out.println("Assertion failed: " + "POS-3" + "start: one two three45" + "EQ" + ___text);
        }
        ___text += String.valueOf(String.valueOf(6) + String.valueOf(" seven"));
        if (!("start: one two three456 seven".equals(___text))) {
            System.out.println("Assertion failed: " + "POS-4" + "start: one two three456 seven" + "EQ" + ___text);
        }
        ___text = String.valueOf(String.valueOf(___text) + String.valueOf(___eight)) + String.valueOf(" nine");
        if (!("start: one two three456 seven eight nine".equals(___text))) {
            System.out.println("Assertion failed: " + "POS-5" + "start: one two three456 seven eight nine" + "EQ" + ___text);
        }
        ___text = String.valueOf(String.valueOf(___text) + String.valueOf("ten")) + String.valueOf(String.valueOf(" eleven") + String.valueOf(" twelve"));
        System.out.println(( ( "start: one two three456 seven eight nine ten eleven twelve".equals(___text) ) ? "String Concat SUCCESS" : "String Concat FAIL" ));
        ___number = Float.parseFloat(___text);
        ___number = (int)(___text.charAt(0));
        ___text = (String.valueOf(10));
        ___text = String.valueOf((char)(int)(30));
        // String Concat -- End --
    }

}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/microbitv2/strings.py
import microbit
import random
import math

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

timer1 = microbit.running_time()

___text = "start:"
___eight = " eight"
___number = 0
def run():
    global timer1, ___text, ___eight, ___number
    # String Concat -- Start --
    ___text += str(" one")
    if not "start: one" == ___text:
        print("Assertion failed: ", "POS-1", "start: one", "EQ", ___text)
    ___text += str("".join(str(arg) for arg in [" two", " three"]))
    if not "start: one two three" == ___text:
        print("Assertion failed: ", "POS-2", "start: one two three", "EQ", ___text)
    ___text += str("".join(str(arg) for arg in [4, 5]))
    if not "start: one two three45" == ___text:
        print("Assertion failed: ", "POS-3", "start: one two three45", "EQ", ___text)
    ___text += str("".join(str(arg) for arg in [6, " seven"]))
    if not "start: one two three456 seven" == ___text:
        print("Assertion failed: ", "POS-4", "start: one two three456 seven", "EQ", ___text)
    ___text = "".join(str(arg) for arg in ["".join(str(arg) for arg in [___text, ___eight]), " nine"])
    if not "start: one two three456 seven eight nine" == ___text:
        print("Assertion failed: ", "POS-5", "start: one two three456 seven eight nine", "EQ", ___text)
    ___text = "".join(str(arg) for arg in ["".join(str(arg) for arg in [___text, "ten"]), "".join(str(arg) for arg in [" eleven", " twelve"])])
    print("String Concat SUCCESS" if ( "start: one two three456 seven eight nine ten eleven twelve" == ___text ) else "String Concat FAIL")
    ___number = float(___text)
    ___number = ord(___text[0])
    ___text = str(10)
    ___text = chr((int)(30))
    # String Concat -- End --

def main():
    try:
        run()
    except Exception as e:
        raise

if __name__ == "__main__":
    main()